File: ros2_ws/src/my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

class MyNode: public rclcpp::Node
{
public:
    MyNode() : Node("cpp_test")
    {
        RCLCPP_INFO(this->get_logger(),"Hello CPP Node");

        timer_ = this->create_wall_timer(std::chrono::seconds(1),
                std::bind(&MyNode::timerCallback, this));
    }
private:

    void timerCallback()
    {
        counter_ ++;
        RCLCPP_INFO(this->get_logger(), "Hello %d", counter_);
    }

    rclcpp::TimerBase::SharedPtr timer_;
    int counter_;
};


int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MyNode>();
    //RCLCPP_INFO(node->get_logger(),"Hello CPP Node");
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: ros2_ws/src/my_py_pkg/my_py_pkg/my_first_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

class MyNode(Node):
	def __init__(self):
		super().__init__("py_test")
		self.counter = 0
		self.get_logger().info("Hello ROS2")
		self.create_timer(0.5, self.timer_callback)

	def timer_callback(self):
		self.counter += 1
		self.get_logger().info("Hello "+str(self.counter))

def main(args=None):
	rclpy.init(args=args)
	#node = Node("py_test")
	#node.get_logger().info("Hello ROS2")
	node = MyNode()
	rclpy.spin(node) 
	rclpy.shutdown()
	
if __name__=="__main__":
	main()
	
